onesixtyone.c

```diff
@@ -132,6 +131,10 @@ void read_communities(char* filename)
 		}
 	}
 
```

Comment:
I think more general and, assuming the below code is correct, very defensive logic for this loop that silently truncates community names that are too long would be the following:
```c
const int max_c = MAX_COMMUNITY_SIZE - 1;
for (i = 0; i < MAX_COMMUNITIES && !feof(fd); ++i) {
  community[i] = (char*) malloc(MAX_COMMUNITY_SIZE);
  community[i][0] = '\0';
  for (c = 0; (ch = fgetc(fd)) != EOF && !isspace(ch); ++c) {
    if (c < max_c) {
      community[i][c] = (char) ch;
      community[i][c + 1] = '\0';
    }
  }
  community[i][max_c] = '\0';
}
```